### ISSUE TITLE
Settings of `addprocs`

### ISSUE THREAD

**Issue by MilesCranmer** [MilesCranmer/SymbolicRegression.jl#180]
I read through the settings of `Distributed.addprocs` in detail to debug some issues I am seeing. I think there are some tweaks that could improve the distributed mode of SymbolicRegression.jl

I think some of the issues noticed by @CharFox1 running in distributed mode are due to the `topology` being incorrectly set to `:worker_to_worker` (the default), rather than `:master_worker` which is what I now think we actually want. The keyword `lazy=true` which fixed this issue actually achieves something similar - it waits to set up channels between workers until they try to communicate with eachother (which never happens, so its like `:master_worker`). So its basically the same thing, but `topology=:master_worker` is technically the correct one we are looking for. e.g., maybe in the future the behavior will actually change.

The other thing we want to tweak is to pass the project in the `exeflags`, rather than running `.activate(...)` after the workers are already set up and connected: https://github.com/MilesCranmer/SymbolicRegression.jl/blob/e07c34681294fdca35a021a0b1a7dc99f334400b/src/Configure.jl#L191-L201

I think (?) this will speed up things for large cluster processing, as running `.activate()` takes a bit of time compared to just passing `--project=...`.

In summary, I think we want to instead be running `addprocs` as follows:

```julia
exeflags = `--project=$(import Pkg: project; splitdir(project().path)[1]) -O3 --startup-file=no`
topology = :master_worker
procs = addprocs_*(
    numprocs;
    exeflags,
    topology,    
)
```

we could also allow the user to pass `addprocs_kws...` to specify other settings/startup tasks.

cc @CharFox1 @patrick-kidger in case of interest

I also posted this as an issue to Julia: https://github.com/JuliaLang/julia/issues/49003 as I think the `--project` setting should really be a default.

**Comment by MilesCranmer** [comment]
Actually, this might not work. It seems like sometimes a worker other than the head worker is the one to wait on a task... So perhaps `lazy=true` is actually the optimal choice!

**Comment by MilesCranmer** [comment]
Apparently Julia 1.9 fixes the environment issue too! So no changes needed after all.
